tools/ansible/tasks/wsu/main.yaml

```diff
@@ -56,3 +56,28 @@
         url: "https://api-int.{{ cluster_address }}:22623/config/worker"
         dest: "{{ win_temp_dir.path }}\\worker.ign"
         validate_certs: no
+
+    - name: Run bootstrapper
+      win_shell: "{{ win_temp_dir.path }}\\wmcb.exe initialize-kubelet --ignition-file {{ win_temp_dir.path }}\\worker.ign --kubelet-path {{ win_temp_dir.path }}\\kubelet.exe"
+      register: bootstrap_out
+
+    - name: Check if bootstrap was successful
+      fail:
+        msg: "Bootstrapper error"
+      when: '"Bootstrapping completed successfully" not in bootstrap_out.stderr'
+
+    - name: Wait for bootstrap CSR
+      pause:
+        seconds: 60
```

Comment:
Instead of a hard coded wait, is it possible to have a loop in the shell code below that retries every 30 seconds and fails after 5 retries?

Reply:
I dont believe so. We currently have no way to tie the CSR back to the node we're bootstrapping.

So we can't wait for a specific CSR to show up.

Reply:
Sure, but can't `oc get csr | awk '/system:serviceaccount:openshift-machine-config-operator:node-bootstrapper/ && /Pending/ {print $1}'` fail with no pending CSRs?

Reply:
Right, its possible that it will return no CSRs. I can make the change to wait for any CSR to show up, but its possible that we will find an unrelated CSR, fulfilling the wait condition.

In that case we will still miss the bootstrapped node's CSR.

I can make that change, but change it to 3 intervals of one minute, in case it takes longer than 30 seconds for the targeted CSR to pop up.

Reply:
>Right, its possible that it will return no CSRs. I can make the change to wait for any CSR to show up, but its possible that we will find an unrelated CSR, fulfilling the wait condition.
In that case we will still miss the bootstrapped node's CSR.

That can happen with the hard coded wait too.

Reply:
Add retries but there is no guarantee that the new node will show up in the updated list even after having some retries